x/auth/vesting/types/vesting_account.go

```diff
@@ -820,8 +821,10 @@ func (va *ClawbackVestingAccount) computeClawback(clawbackTime int64) sdk.Coins
 	totalUnvested := sdk.NewCoins()
 	unvestedIdx := 0
 	for i, period := range va.VestingPeriods {
+		// this period vests at time t, if this occurred before clawback time,
+		// then its already vested.
 		t += period.Length
-		// tie in time goes to clawback
+		// tie in time gets clawed back
 		if t < clawbackTime {
 			totalVested = totalVested.Add(period.Amount...)
 			unvestedIdx = i + 1
```

Comment:
why `i+1`?

Reply:
Veting Periods are stored in slices. E.g) vestingPeriods := types.Period{ {Length: int64(8 * 3600), Amount: c(fee(200))}, {Length: int64(1 * 3600), Amount: c(fee(200))}, {Length: int64(6 * 3600), Amount: c(fee(200))},. Index of these periods indicate the order vesting periods would get processed. Thus, we increment i to store the index of current unvested period in a variable

Reply:
that needs to be written on a comment, it's not clear what the index is used for just by reading the code

Reply:
When the loop exits, `unvestedIdx` will be the index of the first vesting event at or after the clawback time, running past the last element if there is no such event.